contracts/token/ERC20/extensions/ERC20Wrapper.sol

```diff
@@ -18,8 +18,15 @@ import "../utils/SafeERC20.sol";
 abstract contract ERC20Wrapper is ERC20 {
     IERC20 private immutable _underlying;
 
+    /**
+     * @dev The underlying token couldn't be wrapped.
+     */
+    error ERC20InvalidUnderlying(address token);
+
     constructor(IERC20 underlyingToken) {
-        require(underlyingToken != this, "ERC20Wrapper: cannot self wrap");
+        if (underlyingToken == this) {
+            revert ERC20InvalidUnderlying(address(this));
```

Comment:
This error is triggered in the constructor
- so if this is triggered, the contract is never deployed
- so if this is triggered, the contract is not verified

How do we get the ABI to decode the error ? The ethers/truffle instance that tries to do so might have the necessary ABI, but what about etherescan ? I'm concerned the decoding would be an issue.

Maybe we should keep revert strings for things that happens ONLY in constructors ?

Reply:
I asked about this a couple of weeks ago and I think we agreed on not having a special consideration for constructor errors. However I agree with your concerns, but I also think keeping revert strings here wouldn't be the best option (not sure if it increases deployment costs).

I'd keep this since I believe the information is relevant for whoever is deploying using a script or testing, not really useful for someone who's actually deploying, and in such case, spotting the error in the constructor of a contract source code they have access to shouldn't be a problem.